### PR TITLE
Add: initial boilerplate and initial sz.rs (token and line counter)

### PR DESCRIPTION
**Pull request by 21st-centuryman**
Hello Jay, @Jakunot
I implemented sz.rs in a test directory as well as moved all files around from src to src/teenygrad. 

I found the file structure in teenygrad to be annoying to deal with in rust.

We will need to implement sz.rs in github ci/cd. I think that could be something for you if youre ever bored and wanna do something on this project before we officially start in summer. I think this issue also is a great practice for both us of to understand how we request changes, 

I have some small stuff here that I dont think we should add, i leave it up to you to ask me to make changes to this PR.

Cherrio

### CODE REVIEW

**Comment by 21st-centuryman** [comment]
#2

**Comment by 21st-centuryman** [comment]
#3 
Still missing a feature which is listing it by folder
so all tinygrad in one, all nn in one etc.

File: Cargo.toml [hunk 1]
```diff
@@ -4,3 +4,11 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+walkdir = "2"
+tokenizer_py = "0.2.0"
+itertools = "0.12.1"
```
Comment (Jakunot) [inline Cargo.toml:9]: Remove this, not need/ not utilized

**Review by Jakunot — CHANGES_REQUESTED** [review]
Test

File: Cargo.toml (hunk 1 above)
Comment (Jakunot) [inline Cargo.toml:9]: Remove

File: LICENSE
Comment (Jakunot) [inline LICENSE:1]: Revert back

**Review by Jakunot — CHANGES_REQUESTED** [review]
Fix suggested changes

File: src/tests/sz.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+#[test]
+fn lines() -> std::io::Result<()> {
+    let token_whitelist = |token: &Token| {
+        matches!(
+            token,
+            Token::OP(_) | Token::Name(_) | Token::Number(_) | Token::String(_)
+        )
+    };
+    let _headers = vec!["Name", "Lines", "Tokens/Line"];
```
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply src/tests/sz.rs]: [065a5e6](https://github.com/teeny-kth/teenygrad/pull/1/commits/065a5e6873a896d8a12576812ed5cba17ad4fb55)

File: src/teenygrad/tensor.rs
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply src/teenygrad/tensor.rs:1]: [065a5e6](https://github.com/teeny-kth/teenygrad/pull/1/commits/065a5e6873a896d8a12576812ed5cba17ad4fb55)

File: src/teenygrad/lazy.rs
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply src/teenygrad/lazy.rs:1]: [065a5e6](https://github.com/teeny-kth/teenygrad/pull/1/commits/065a5e6873a896d8a12576812ed5cba17ad4fb55)

File: src/teenygrad/function.rs
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply src/teenygrad/function.rs:1]: [065a5e6](https://github.com/teeny-kth/teenygrad/pull/1/commits/065a5e6873a896d8a12576812ed5cba17ad4fb55)

File: LICENSE
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply LICENSE:1]: [065a5e6](https://github.com/teeny-kth/teenygrad/pull/1/commits/065a5e6873a896d8a12576812ed5cba17ad4fb55)

File: src/teenygrad/ops.rs
[the remark this answers is not in the document]
Comment (21st-centuryman) [reply src/teenygrad/ops.rs:1]: [fc8f1f8](https://github.com/teeny-kth/teenygrad/pull/1/commits/fc8f1f8602d18d9b4a25e67a5e1a18745a218c5b)

File: Cargo.toml (hunk 1 above)
Comment (21st-centuryman) [reply Cargo.toml]: [fc8f1f8](https://github.com/teeny-kth/teenygrad/pull/1/commits/fc8f1f8602d18d9b4a25e67a5e1a18745a218c5b)

**Review by Jakunot — APPROVED** [review]
lgtm

**Pull request merged**